### PR TITLE
Update `output.md`

### PR DESCRIPTION
**Pull request by anlavandier**
This PR updates `output.md` in light of the recent changes in the classes `OutputManager` and `PostProcessManager`.

This PR also fixes a small file extension related issue in `psydac/api/postprocessing.py` and ghost regions issues in the same file.

### CODE REVIEW

File: output.md [hunk 1]
```diff
@@ -90,59 +95,73 @@ file.h5
...
         attribute: t
-        attribute: ts 
+        attribute: ts
         space_1/
         ...
         space_n/
     ...
     snapshot_n/
 ```
-In addition to that, psydac also has a class to read those files, recreate all the `FemSpace` and `FemField` objects and export them to `VTK`.
```
Comment (yguclu) [inline output.md:100]: - Capitalize: psydac -> Psydac
- Please add class name: PostProcessManager

File: output.md [hunk 2]
```diff
@@ -90,59 +95,73 @@ file.h5
...
+References to the available FEM spaces are given to the OutputManager object through the `add_spaces(**kwargs)` method, and the corresponding YAML file is created upon calling the method `export_spaces_info()`. In order to inform the OutputManager object that the next fields to be exported are time-independent, the user should call the `set_static()` method. In the case of time-dependent fields, the user should prepare a time snapshot (which is defined for a specific time step `ts` and time value `t`) by calling the method `add_snapshot(t, ts)`. In both cases the fields are exported to the HDF5 file through a call to the method `export_fields(**kwargs)`. Here is a usage example:
```
Comment (yguclu) [inline output.md:114]: - Typo: export_spaces_info -> export_space_info (singular)
- If you do not mention that `ts` is an index, you should at least say that it is an integer

File: output.md [hunk 3]
```diff
@@ -90,59 +95,73 @@ file.h5
...
+4.  `snapshots` specifies which time snapshots should be extracted from the HDF5 file (`none` in the case of static fields)
+5.  `fields` is a tuple  of `h5_field_name`.
+
+Here is a usage example:
 
 ```python
 # geometry.h5 is where the domain comes from. See PostProcessManager's docstring for me information
 post = PostProcessManager(geometry_file='geometry.h5', space_file='spaces.yml', fields_file='fields.h5')
 
 # See PostProcessManager.export_to_vtk's and TensorFemSpace.eval_fields' doscstrings for more information
```
Comment (yguclu) [inline output.md:157]: Typo: doscstrings -> docstrings

File: output.md [hunk 4]
```diff
@@ -90,59 +95,73 @@ file.h5
...
-
-output_m.export_spaces_info() # Writes the space information to Yaml
 ```
 
 ## Usage of class `PostProcessManager`
 
 Typically the `PostProcessManager` class is used in a separate post-processing script, which is run after the simulation has finished. In essence it evaluates the FEM fields over a uniform grid (applying the appropriate push-forward operations) and exports the values to a VTK file (or a sequence of files in the case of a time series). An instance of the `PostProcessManager` class is created by specifying the following:
 
-1.  The name of the geometry file (in HDF5 format) which defines the multi-patch geometry of interest
+1.  The name of the geometry file or the topological domain
```
Comment (yguclu) [inline output.md:140]: I think this point needs a bit more explanation. Could you not keep (some of) the original text?

File: psydac/api/postprocessing.py [hunk 5]
```diff
@@ -187,8 +186,8 @@ def __init__(self, filename_space, filename_fields, comm=None, mode='w', save_mp
         self._space_info = {}
         self._spaces = []
 
-        if filename_space[-4:] != ".yml" and filename_space[-4:] != ".yaml":
-            filename_space += ".yml"
+        if not filename_space.split('.')[-1] in ['yml', 'yaml']:
```
Comment (yguclu) [inline psydac/api/postprocessing.py:189]: Can you not use `os.path.splitext`? See the [docs](https://docs.python.org/3/library/os.path.html#os.path.splitext)

Using`pathlib` is also possible, for a more object-oriented approach:
```python
from pathlib import Path
...

if Path(filename_space).suffix not in ('.yml', '.yaml'):
    ...
```

File: psydac/api/postprocessing.py [hunk 6]
```diff
@@ -187,8 +186,8 @@ def __init__(self, filename_space, filename_fields, comm=None, mode='w', save_mp
...
+            filename_space += ".yaml"
         self.filename_space = filename_space
 
         if filename_fields[-3:] != ".h5":
```
Comment (yguclu) [inline psydac/api/postprocessing.py:193]: Same comment

File: psydac/api/postprocessing.py [hunk 7]
```diff
@@ -722,7 +721,7 @@ def __init__(self, geometry_file=None, domain=None, space_file=None, fields_file
         self._domain_h = None
 
         if not space_file.split('.')[-1] in ['yml', 'yaml']:
```
Comment (yguclu) [inline psydac/api/postprocessing.py:723]: Same comment

**Review by yguclu — CHANGES_REQUESTED** [review]
I've just noticed a typo in the title of file `output.md`: ouputs -> outputs

File: output.md (hunk 1 above)
Comment (anlavandier) [reply output.md:100]: Fixed in [34bfd98](https://github.com/pyccel/psydac/pull/234/commits/34bfd9845a7a7c44d8589ff9b670b919f204d457)

File: output.md (hunk 2 above)
Comment (anlavandier) [reply output.md]: Fixed in [34bfd98](https://github.com/pyccel/psydac/pull/234/commits/34bfd9845a7a7c44d8589ff9b670b919f204d457)

File: output.md (hunk 3 above)
Comment (anlavandier) [reply output.md]: Fixed in [34bfd98](https://github.com/pyccel/psydac/pull/234/commits/34bfd9845a7a7c44d8589ff9b670b919f204d457)

File: output.md (hunk 4 above)
Comment (anlavandier) [reply output.md]: Put back some of the text in [34bfd98](https://github.com/pyccel/psydac/pull/234/commits/34bfd9845a7a7c44d8589ff9b670b919f204d457)

File: psydac/api/postprocessing.py (hunk 5 above)
Comment (anlavandier) [reply psydac/api/postprocessing.py]: Used `os.path.splitext` in [fba77de](https://github.com/pyccel/psydac/pull/234/commits/fba77de0bc83157d27c5f5bbf7f3fd75a05dc210)

File: psydac/api/postprocessing.py (hunk 6 above)
Comment (anlavandier) [reply psydac/api/postprocessing.py]: Same as above. [fba77de](https://github.com/pyccel/psydac/pull/234/commits/fba77de0bc83157d27c5f5bbf7f3fd75a05dc210)

File: psydac/api/postprocessing.py (hunk 7 above)
Comment (anlavandier) [reply psydac/api/postprocessing.py]: Same as above. [fba77de](https://github.com/pyccel/psydac/pull/234/commits/fba77de0bc83157d27c5f5bbf7f3fd75a05dc210)

File: psydac/api/postprocessing.py [hunk 8]
```diff
@@ -720,12 +724,12 @@ def __init__(self, geometry_file=None, domain=None, space_file=None, fields_file
...
         self._domain_h = None
 
-        if not space_file.split('.')[-1] in ['yml', 'yaml']:
+        if not os.path.splitext(space_file)[-1] in ['.yml', '.yaml']:
             self.space_filename = space_file + '.yaml'
         else:
             self.space_filename = space_file
 
-        if fields_file.split('.')[-1] != 'h5':
+        if not os.path.splitext(fields_file)[-1] != 'h5':
```
Comment (yguclu) [inline psydac/api/postprocessing.py:732]: I think there should be no `not` here

File: psydac/api/postprocessing.py [hunk 9]
```diff
@@ -1311,10 +1311,12 @@ def _load_fields_helper(self, patch_name, space_name, field_name, coeff, contain
                 V = space.spaces[i].vector_space
                 index_coeff = tuple(slice(s, e + 1) for s, e in zip(V.starts, V.ends))
                 field.coeffs[i][index_coeff] = coeff[i][index_coeff]
+                field.coeffs[i].update_ghost_regions()
         else:
             V = space.vector_space
             index_coeff = tuple(slice(s, e + 1) for s, e in zip(V.starts, V.ends))
             field.coeffs[index_coeff] = coeff[index_coeff]
+            field.coeffs.update_ghotst_regions()
```
Comment (yguclu) [inline psydac/api/postprocessing.py:1319]: Typo!

**Review by yguclu — CHANGES_REQUESTED** [review]
As I said before, the title of the file (not its name!) contains a typo...

**Comment by anlavandier** [comment]
> As I said before, the title of the file (not its name!) contains a typo...

Fixed in [e3bf69a](https://github.com/pyccel/psydac/pull/234/commits/e3bf69adcf4b6a855c482add7225a8f5a779a536)

File: psydac/api/postprocessing.py (hunk 9 above)
Comment (anlavandier) [reply psydac/api/postprocessing.py]: Fixed in [18a090c](https://github.com/pyccel/psydac/pull/234/commits/18a090c4864587c5a3f951cfdec613fedbc124d4)

File: psydac/api/postprocessing.py (hunk 8 above)
Comment (anlavandier) [reply psydac/api/postprocessing.py]: Fixed in [18a090c](https://github.com/pyccel/psydac/pull/234/commits/18a090c4864587c5a3f951cfdec613fedbc124d4)

**Review by yguclu — CHANGES_REQUESTED** [review]
If you don't mind, please restore the file name to the original `output.md` (singular).

Also, could you please reference this file from the global README?

**Review by yguclu — APPROVED** [review]
Great job!!

**Pull request merged**